lib/sentry/transport.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package sentry
+
+import (
+	"github.com/getsentry/raven-go"
+	"github.com/pkg/errors"
+)
+
+var ErrSentryDisabled = errors.New("sentry disabled via config")
```

Comment:
I think all the raven stuff becomes no-ops if the DSN is empty?

Reply:
Sadly not, it just does nothing (does not unset the DSN):

https://github.com/getsentry/raven-go/blob/master/client.go#L488

Reply:
That's fascinating, given https://docs.sentry.io/clients/go/config/

```
SetDSN

The DSN tells the SDK where to send the events. This option is always required
and the Sentry Go SDKs can pick this up from an environment variable (SENTRY_DSN).
When set to an empty string, SDK won’t send any events to Sentry, and all Capture*
methods will effectively act as no-ops.
```